src/connection-manager/dialer/dial-request.ts

```diff
@@ -86,10 +86,10 @@ export class DialRequest {
         const token = await tokenHolder.shift() // get token
         let conn
         try {
-          const signal = dialAbortControllers[i].signal
+          const signal = (dialAbortControllers[i] as AbortController).signal
```

Comment:
Instead of casting, please null-guard on the abort controller instead.

Reply:
changed